web-app/js/portal/data/MetadataRecordFetcher.js

```diff
@@ -34,10 +34,11 @@ Portal.data.MetadataRecordFetcher = Ext.extend(Ext.util.Observable, {
             var store = new Portal.data.MetadataRecordStore();
             store.loadData(response.responseXML);
             var metadataRecord = store.getAt(0);
-            var dataCollection = Portal.data.DataCollection.fromMetadataRecord(metadataRecord);
 
             if (metadataRecord) {
+                var dataCollection = Portal.data.DataCollection.fromMetadataRecord(metadataRecord);
                 _this.dataCollectionStore.add(dataCollection);
+
```

Comment:
Because loading a DataCollection and adding it to the store is a separate thing from setting the tab in a UI component (the line below). So I separated them.